src/gamma.jl

```diff
@@ -692,22 +709,24 @@ function lgamma(z::Complex{Float64})
     else
         shift = Complex(real(shift), imag(shift) + signflips*6.2831853071795864769252842)
     end
-    return lgamma_asymptotic(Complex(x,y)) - shift
+    return loggamma_asymptotic(Complex(x,y)) - shift
 end
-lgamma(z::Complex{T}) where {T<:Union{Integer,Rational}} = lgamma(float(z))
-lgamma(z::Complex{T}) where {T<:Union{Float32,Float16}} = Complex{T}(lgamma(Complex{Float64}(z)))
+loggamma(z::Complex{T}) where {T<:Union{Integer,Rational}} = loggamma(float(z))
+loggamma(z::Complex{T}) where {T<:Union{Float32,Float16}} = Complex{T}(loggamma(Complex{Float64}(z)))
+loggamma(x::Float16) = Float16(loggamma(Float32(x)))
```

Comment:
This one can be deleted as well
```suggestion
```

Reply:
There are some previous tests like `loggamma(x) ≈ zero(Float16)` which will fail then. Should I remove those?

Reply:
no, this should be caught by the `loggamma(x::Real)` method.

Reply:
Test threw exception
  Expression: loggamma(x) ≈ zero(Float16)
  MethodError: no method matching Float16(::Tuple{Float32,Int32})
  Closest candidates are:
    Float16(!Matched::BigInt) at gmp.jl:391
    Float16(!Matched::Integer) at float.jl:51
    Float16(!Matched::Float32) at float.jl:141
    ...
  Stacktrace:
   [1] logabsgamma(::Float16) at /home/sumegh/.julia/packages/SpecialFunctions/fvheQ/src/gamma.jl:800
   [2] loggamma(::Float16) at /home/sumegh/.julia/packages/SpecialFunctions/fvheQ/src/gamma.jl:625
   [3] top-level scope at /home/sumegh/.julia/packages/SpecialFunctions/fvheQ/test/runtests.jl:622
   [4] include at ./boot.jl:317 [inlined]
   [5] include_relative(::Module, ::String) at ./loading.jl:1044
   [6] include(::Module, ::String) at ./sysimg.jl:29
   [7] include(::String) at ./client.jl:392
   [8] top-level scope at none:0
   [9] eval(::Module, ::Any) at ./boot.jl:319
   [10] macro expansion at ./logging.jl:317 [inlined]
   [11] exec_options(::Base.JLOptions) at ./client.jl:219
   [12] _start() at ./client.jl:425
Other float types: